addons/minio/template/base/install.sh

```diff
@@ -10,10 +10,290 @@ function minio_pre_init() {
+# minio deployment then copies the object buckets from the old deployment into the new one. this function generates
+# unavailability of the original minio service.
+function migrate_fs_backend() {
+    # if minio is not running then we can't migrate its data. minio may be deployed by means of a statefulset as well
+    # but on this case the running pods are guaranteed to not leverage the fs backend.
+    local minio_replicas
+    minio_replicas=$(kubectl get --no-headers deploy minio -n $MINIO_NAMESPACE -o custom-columns=:.spec.replicas 2>/dev/null)
+    if [ -z "$minio_replicas" ] || [ "$minio_replicas" = "0" ]; then
+        return
+    fi
+
+    # if minio does not use fs backend we don't need to migrate it.
+    if ! minio_uses_fs_format ; then
+        return
+    fi
+
+    printf "${YELLOW}\n"
+    printf "The installer has detected that the cluster is running a version of minio backed by the now legacy FS format.\n"
+    printf "To be able to upgrade to the new Minio version a migration will be necessary. During this migration, Minio\n"
+    printf "will be unavailable.\n"
+    printf "\n"
+    printf "For further information please check https://github.com/minio/minio/releases/tag/RELEASE.2022-10-29T06-21-33Z\n"
+    printf "${NC}\n"
+    printf "Would you like to proceed with the migration ?"
+    if ! confirmN; then
+        bail "Not migrating"
+    fi
+
+    local minio_access_key
+    minio_access_key=$(kubernetes_secret_value $MINIO_NAMESPACE minio-credentials MINIO_ACCESS_KEY)
+    if [ -z "$minio_access_key" ]; then
+        bail "failed to read minio access key"
+    fi
+
+    local minio_secret_key
+    minio_secret_key=$(kubernetes_secret_value $MINIO_NAMESPACE minio-credentials MINIO_SECRET_KEY)
+    if [ -z "$minio_secret_key" ]; then
+        bail "failed to read minio access key"
+    fi
+
+    local minio_pod_ip
+    minio_pod_ip=$(kubectl get pods -n $MINIO_NAMESPACE -l app=minio --no-headers -o custom-columns=:.status.podIP | head -1)
+    if [ -z "$minio_pod_ip" ]; then
+        bail "failed to determine minio pod ip address"
+    fi
+
+    disable_minio_svc
+    if ! spinner_until 120 svc_has_no_endpoints $MINIO_NAMESPACE "minio" ; then
+        enable_minio_svc
+        bail "timeout waiting for minio service to be decomissioned"
+    fi
+
+    if ! create_fs_migration_deployment ; then
+        enable_minio_svc
+        bail "failed to start minio migration deployment"
+    fi
+
+    if ! migrate_object_store \
+        "$MINIO_NAMESPACE" \
+        "$minio_pod_ip:9000" \
+        "$minio_access_key" \
+        "$minio_secret_key" \
+        "minio-migrate-fs-backend" \
+        "$minio_access_key" \
+        "$minio_secret_key"
+    then
+        enable_minio_svc
+        destroy_fs_migration_deployment
+        bail "failed to migrate data to minio migration deployment"
+    fi
+
+    # scale down minio and wait until it is out of service.
+    kubectl scale --replicas=0 deployment minio -n $MINIO_NAMESPACE
```

Comment:
```suggestion
    kubectl scale --replicas=0 deployment minio -n "$MINIO_NAMESPACE"
```

Reply:
i personally think this reads better like the following. i tend to prefer the namespace as the first parameter after kubectl

```suggestion
    kubectl -n "$MINIO_NAMESPACE" scale deployment minio --replicas=0
```